flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/sharedbuffer/Lockable.java

```diff
@@ -109,8 +110,10 @@ public boolean isImmutableType() {
 		}
 
 		@Override
-		public TypeSerializer<Lockable<E>> duplicate() {
-			return new LockableTypeSerializer<>(elementSerializer);
+		public LockableTypeSerializer<E> duplicate() {
```

Comment:
why are you changing the signature?

Reply:
Because some code might want to get a `LockableTypeSerializer` when duplicating a `LockableTypeSerializer` and that also only happens when you already have a hold on a `LockableTypeSerializer` otherwise the interface will still give you the general type. A more practical example, the test required this to access the `getElementSerializer()` method.